components/playlist/browser/resources/components/contextualMenu.tsx

```diff
@@ -47,18 +47,25 @@ export default function ContextualMenuAnchorButton ({
   onDismissMenu
 }: MenuProps) {
   const menuButtonProps: ButtonMenuProps = {}
-  // Force menu widget to be closed when the anchor button is not visible. In
-  // case where it's visible, the menuButtonProps doesn't contain isOpen property,
-  // so it won't affect the behavior of the menu.
-  if (!visible) menuButtonProps.isOpen = false
+  if (visible) {
```

Comment:
Generally its not a good idea to change between a controlled and uncontrolled version of a `Component` in React (for example, an `input` will complain if you change it's value from `undefined` to the `''` and vice versa).

If we're managing the `open` state, it's normally better to be explicit about it:

```tsx
const [open, setOpen] = useState(false)

// When the anchor button isn't visible, hide the menu
useEffect(() => !visible && setOpen(false), [visible])
return <StyledButtonMenu
    tabIndex={0}
    visible={visible}
    onChange={({isOpen }) => {
      if (isOpen) showMenu?.()
      setOpen(isOpen)
    }}
    onClose={() => onDismissMenu?.()}
    open={isOpen} >

Reply:
Awesome! Done. (Slightly changed due to compiler complains about the return type of the `!visible && setOpen()`.